feat/gene/gene.go

```diff
@@ -285,6 +314,38 @@ func (t *CodingTranscript) SetExons(exons ...Exon) error {
 	return nil
 }
 
+// TranscriptFeature defines a feature on a transcript.
+type TranscriptFeature struct {
+	Transcript Transcript
+	Offset     int
+	Length     int
+	Orient     feat.Orientation
+	Desc       string
+}
+
+// Start returns the feature start relative to Transcript.
+func (t *TranscriptFeature) Start() int { return t.Offset }
+
+// End returns the feature end relative to TranscriptLocation.
+func (t *TranscriptFeature) End() int { return t.Offset + t.Length }
+
+// Len returns the length of the feature.
+func (t *TranscriptFeature) Len() int { return t.Length }
+
+// Name returns an empty string.
```

Comment:
I couldn't think of a good reason to add an extra field for name so I followed what we did for Exon below. I will add it now.